metrics/src/main/java/org/wildfly/extension/metrics/WildFlyMetric.java

```diff
@@ -36,10 +36,11 @@
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
 import org.jboss.dmr.ModelNode;
-import org.jboss.dmr.ModelType;
 
 public class WildFlyMetric implements Metric {
 
+    private static final ModelNode UNDEFINED = new ModelNode();
```

Comment:
Please add a static block:

static {
    UNDEFINED.protect();
}

That makes the node immutable and thus safe to pass around.

Reply:
Ah. Very cool. Done.